docs/IR.md

```diff
@@ -282,6 +282,9 @@ There are two ways to leave an optional input or output unspecified: the first,
 
 Each node referring to an operator with optional outputs MUST provide a name for each output that is computed and MUST NOT provide names for outputs that are not computed.
 
+#### External Tensor Data
+
+The raw data for large constant tensors, such as initializers, MAY be serialised in a separate file. In such a case, the tensor MUST provide the filename relative to the model file and MUST NOT use the value fields. It MAY provide a byte offset and length within that file. It MAY also specify a SHA1 digest of the file. One file MAY contain the data for multiple tensors.
 
```

Comment:
However, this feature is experimental, it may break shape inference.  And we're not going to fix it.

Reply:
I cannot find anything saying external raw data is experimental in code/docs/PR comments, so if that's true then documenting its experimental status should be high priority.  
AFAICT, external data is supported at least by the ONNX Python API (onnx/onnx#678) and ONNX Runtime (microsoft/onnxruntime#520 e.g. test file `opset9/tf_resnet_v2_50/model.onnx`).

I wouldn't expect it to break type inference, since the type/shape is stored in the [`data_type`/`dims` fields](https://github.com/onnx/onnx/blob/master/onnx/onnx.in.proto#L351-L356) on TensorProto, not the `raw_data` field.

Apologies if I misunderstood something.

Reply:
It can breaks shape inference, for example, when `Reshape`'s shape is stored in external data.

Reply:
Okay, that makes sense.

Maybe a dumb question, but why are tensors in external files considered inaccessible for shape inference?

Reply:
It could, but it need a big effort.